Example/SBTUITestTunnel_Tests/StubTests.swift

```diff
@@ -74,34 +74,43 @@ class StubTests: XCTestCase {
         XCTAssert(request.headers(headers, isEqual: expectedHeaders))
     }
     
-    func testStubAddTwice() {
-        // first rule should win
-        app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 1]))
-        app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["not-stubbed": 99]))
-        
-        let result = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
-        XCTAssert(request.isStubbed(result))
+    func testMultipleStubsForSameRequestMatch() {
+        XCTContext.runActivity(named: "When adding multiple stubs for the same requests match") { _ in
+            app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 1], returnCode: 200))
+            app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 1], returnCode: 401, activeIterations: 1))
+            app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 1], returnCode: 500, activeIterations: 2))
+        }
         
-        XCTAssert(app.stubRequestsRemoveAll())
-        let result2 = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
-        XCTAssertFalse(request.isStubbed(result2))
+        XCTContext.runActivity(named: "They are evaluated in LIFO order and removed when finishing active iterations") { _ in
+            let result = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
+            XCTAssertEqual(request.returnCode(result), 500)
+            
+            let result2 = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
+            XCTAssertEqual(request.returnCode(result2), 500)
+            
+            let result3 = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
+            XCTAssertEqual(request.returnCode(result3), 401)
+
+            let result4 = request.dataTaskNetwork(urlString: "http://httpbin.org/get?param1=val1&param2=val2")
+            XCTAssertEqual(request.returnCode(result4), 200)
+        }
     }
     
     func testStubAddTwiceAndRemovedOnce() {
-        // first rule should win
-        let stubId1 = app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 1])) ?? ""
-        let stubId2 = app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["not-stubbed": 99])) ?? ""
+        // Rules are evaluated in a LIFO order
+        let stubId1 = app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["not-stubbed": 99])) ?? ""
```

Comment:
Could you please rebase with master? I've modified the `isStubbed` method so that you can check also the stub value.

I would expect this test to be rewritten like
```suggestion
        let stubId1 = app.stubRequests(matching: SBTRequestMatch(url: "httpbin.org"), response: SBTStubResponse(response: ["stubbed": 2])) ?? ""
```

so that you can first check that you really check the stubbing LIFO order. As is you're not sure that the `stubId1` is actually being used.

Reply:
I would then add 2 tests that verify the same for rewrite and cookie block (so that you test LIFO order in lines SBTProxyURLProtocol:378-386